### PR TITLE
Randomized Search CV in ML

### LINKED ISSUE #769: RandomSearchCV in Machine Learning
I will explain the concept of RandomSearchCV which is used to find the best parameters for a model to train. Please assign this issue to me

### PR DESCRIPTION
**Pull request by Rithish5513U**
## PR Description

This PR is raised to merge RandomizedSearchCV in ML which is used to optimize the loss function in ML models with hyperparameter tuning.

## Related Issues

- Closes #769 

### Checklist

- [x] I have gone through the [contributing guide](https://github.com/animator/learn-python/blob/main/CONTRIBUTING.md)
- [x] I have updated my branch and synced it with project `main` branch before making this PR

## Undertaking

I declare that: 

1. The content I am submitting is original and has not been plagiarized.  
2. No portion of the work has been copied from any other source without proper attribution.  
3. The work has been checked for plagiarism, and I assure its authenticity.  

I understand that any violation of this undertaking may have legal consequences that I will bear and could result in the withdrawal of any recognition associated with the work. 

- [x] I Agree

### CODE REVIEW

**Review by SurajSanap — APPROVED** [review]
Approve
